### PR TITLE
HBHE M3 and timeSlew

### PR DESCRIPTION
**Pull request by mariadalfonso**
This PR add add a switch to disable the timeslew for the online HCAL energy (Method3).
Added dedicated siPM delay parameters (for now they contains zero).

changes expected only in the 2017 scenario and for HE only

Validation plots here
http://dalfonso.web.cern.ch/dalfonso/ChangesInM3.pdf

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
@cmsbuild please test
the last test fell into the gap of partially not working comparisons

**Comment by slava77** [comment]
@mariadalfonso 
what is the size of expected changes?
Jenkins comparison tests are showing no differences.
- HLT table in 2017 wflow is fake (~ no-op) and no changes are expected here
- In pre-phase1 workflows M3 results are saved in eaux in regular reco, but the 2017 SimpleHBHEPhase1Algo does not runM3 by default (in addition to M2). So, all saved values are 0.

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/16259 2b46bf2
- changes are in line with the description. Logically, the pre-phase-1 workflows are not affected; the M3 values are not computed in existing 2017 workflows yet.
- jenkins tests pass and comparisons with baseline show no difference as expected

**Comment by kpedro88** [comment]
@mmusich please sign, only AlCa-relevant item is minor change in HcalTimeSlew

**Comment by mmusich** [comment]
+1
I presume the parameters for SiPM at https://github.com/cms-sw/cmssw/pull/16259/files#diff-86747d308c7c97710dbbea2bb292dc96R11 will be changed later on.

File: CalibCalorimetry/HcalAlgos/src/HcalTimeSlew.cc [hunk 1]
```diff
@@ -8,13 +8,16 @@ static const double cap = 6.0;
...
+static const double tspar1_siPM[2] = {0, 0};
+static const double tspar2_siPM[2] = {0, 0};
 
 double HcalTimeSlew::delay(double fC, BiasSetting bias) {
   double rawDelay=tzero[bias]+slope[bias]*log(fC);
   return (rawDelay<0)?(0):((rawDelay>tmax[bias])?(tmax[bias]):(rawDelay));			   
 }
 
-double HcalTimeSlew::delay(double fC, ParaSource source, BiasSetting bias, double par0, double par1, double par2) {
+double HcalTimeSlew::delay(double fC, ParaSource source, BiasSetting bias, double par0, double par1, double par2, bool isHPD) {
```
Comment (davidlange6) [inline CalibCalorimetry/HcalAlgos/src/HcalTimeSlew.cc]: Hi @mariadalfonso @slava77 - as the complexity of this grows, maybe something that does more at initialization time will be more appropriate

**Comment by davidlange6** [comment]
+1

File: CalibCalorimetry/HcalAlgos/src/HcalTimeSlew.cc (hunk 1 above)
Comment (slava77) [inline CalibCalorimetry/HcalAlgos/src/HcalTimeSlew.cc]: @davidlange6 
please clarify what you mean.
Both HPD and SiPM hits are in the same input collection.

Another alternative is to create two instances of the hit processing code configured appropriately and then swap out the appropriate one at the right point.
There is no single point of entry for all hit processing algorithms. So, percolation of this alternative will need some refactoring.
Is this the way you suggest to go?

@igv4321

**Pull request merged**